### ISSUE TITLE
Consider suppressing `question_mark` lint if `question_mark_used` restriction is being used

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#11283]
### Summary

When using `deny(clippy::question_mark_used)`, it would be convenient not to have to explicitly put `allow(clippy::question_mark)`, a lint whose sole purpose is to recommend using `?`.

I don't know how feasible this is to implement &mdash; I can imagine it might be architecturally hard for a lint to be sensitive to the setting of some other lint. Feel free to close in this case.

### Lint Name

question_mark

### Reproducer

```rust
#![deny(clippy::question_mark_used)]

fn f() -> Result<(), String> {
    Ok(())
}

fn main() -> Result<(), String> {
    if let Err(err) = f() {
        return Err(err);
    }
    Ok(())
}
```

```console
$ cargo clippy

warning: this block may be rewritten with the `?` operator
  --> src/main.rs:8:5
   |
8  | /     if let Err(err) = f() {
9  | |         return Err(err);
10 | |     }
   | |_____^ help: replace it with: `f()?;`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#question_mark
   = note: `#[warn(clippy::question_mark)]` on by default
```

### Version

```text
rustc 1.73.0-nightly (d12c6e947 2023-08-01)
binary: rustc
commit-hash: d12c6e947ceacf3b22c154caf9532b390d8dc88a
commit-date: 2023-08-01
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5
```
